label_studio_ml/examples/yolo/README_TIMELINE_LABELS.md

```diff
@@ -0,0 +1,517 @@
+- **YOLO model limitations**: The model uses a pre-trained YOLO model trained on object classification tasks for feature extraction, which may not be optimal for all use cases such as event detection. This approach doesn't tune the YOLO model, it trains only the LSTM part upon.
+- **Label balance**: The model may struggle with imbalanced labels. Ensure that the labels are well-distributed in the training data. Consider modifying the loss function (BCEWithLogitsLoss) and using class pos weights to address this issue.
+- **Training on all daa**: Training on all data is not yet implemented, so the model trains only on the last annotation. See `timeline_labels.py::fit()` for more details. 
+
+## Example use case: detecting a ball in football videos
+
+### Setup
+
+1. **Labeling Configuration**:
+
+   ```xml
+   <View>
+      <TimelineLabels name="videoLabels" toName="video">
+        <Label value="Ball touch" background="red"/>
+        <Label value="Ball in frame" background="blue"/>
+      </TimelineLabels>
+      <Video name="video" value="$video" height="700" timelineHeight="200" frameRate="25.0" />
+    </View>
+   ```
+
+2. **Connect the Model Backend**:
+
+   - Create a new project, go to Project Settings > Model. Add the YOLO backend to your project.
+     1. navigate to the yolo folder in this repository in your terminal.
+     2. Update your `docker-compose.yml` file
+     3. Run `docker compose up` to run the backend
+     4. Connect this backend to your LabelStudio project in the project settings. Make sure that Interactive Preannotations is OFF (this is the default). 
+
+### Annotation and training
+
+1. **Annotate Videos**:
+
+   - Upload football videos to the project.
+   - Use the TimelineLabels control tag to label time intervals where the ball is visible in the frame.
+
+2. **Model Training**:
+
+   - After submitting annotations, the model begins training incrementally.
+   - Continue annotating until the model starts making accurate predictions.
+
+3. **Review Predictions**:
+
+   - The model suggests labels for unannotated videos.
+   - Validate and correct predictions to further improve the model.
+
+## Adjusting training parameters and resetting the model
+
+If the model is not performing well, consider modifying the LSTM and classifier training parameters in the labeling config,
+they start with `model_classifier_` prefix.
+
+The model will be **reset** after changing these parameters:
+ - model_classifier_sequence_size,
+ - model_classifier_hidden_size,
+ - model_classifier_num_layers,
+ - new labels added or removed from the labeling config.
+
+So, you may need to re-update (click the "Update" button) on annotations to see improvements.
+
+If you want to modify more parameters, you can do it directly in the code in the `utils/neural_nets.py::MultiLabelLSTM`.
+
+If you need to reset the model completely, you can remove the model file from `/app/models`. 
```

Comment:
I think this folder only exists in the Docker container

Reply:
I would like to get access of cache_dir and models for users outside of docker. They should be able to manage it